app/controllers/shopify/callbacks_controller.rb

```diff
@@ -0,0 +1,74 @@
+class Shopify::CallbacksController < ApplicationController
+  include Shopify::IntegrationHelper
+
+  def show
+    verify_account!
+
+    @response = oauth_client.auth_code.get_token(
+      params[:code],
+      redirect_uri: '/shopify/callback'
+    )
+
+    handle_response
+  rescue StandardError => e
+    Rails.logger.error("Shopify callback error: #{e.message}")
+    redirect_to "#{redirect_uri}?error=true"
+  end
+
+  private
+
+  def verify_account!
+    @account_id = verify_shopify_token(params[:state])
+    raise StandardError, 'Invalid state parameter' if account.blank?
+  end
+
+  def handle_response
+    account.hooks.create!(
+      app_id: 'shopify',
+      access_token: parsed_body['access_token'],
+      status: 'enabled',
+      reference_id: params[:shop],
+      settings: {
+        token_type: parsed_body['token_type'],
+        expires_in: parsed_body['expires_in'],
```

Comment:
When does this token expire? If it does expire, do we need to refresh the token?

Reply:
offline access token doesn't expire AFAIK. I just removed this. These values are empty in the response.